server/main.lua

```diff
@@ -61,8 +73,9 @@ RegisterNetEvent('qbx_diving:server:takeCoral', function(coralIndex)
     exports.ox_inventory:AddItem(src, coralType.item, amount)
     pickedUpCoralIndexes[coralIndex] = true
     TriggerClientEvent('qbx_diving:client:coralTaken', -1, coralIndex)
-    TriggerEvent('qbx_diving:server:coralTaken', sharedConfig.coralLocations[currentAreaIndex].corals[coralIndex].coords)
-    if #pickedUpCoralIndexes == sharedConfig.coralLocations[currentAreaIndex].maxHarvestAmount then
```

Comment:
This method is still preferred over adding an unnecessary local function

Reply:
This wasn't working. #pickedUpCoralIndexes wasn't counting pickedUpCoralIndexes[coralIndex] = true. 

There's probably a better way to fix this compared to what I did, but it's what I knew.

Reply:
Use https://docs.qbox.re/resources/qbx_core/modules/lib/shared#tablesize